hudi-common/src/main/java/org/apache/hudi/common/fs/FSUtils.java

```diff
@@ -780,6 +780,22 @@ public static <T> Map<String, T> parallelizeFilesProcess(
     return result;
   }
 
+  public static <T, R> Map<T, R> parallelizeProcess(
+      HoodieEngineContext hoodieEngineContext,
+      int parallelism,
+      SerializableFunction<T, R> pairFunction,
```

Comment:
Seems a general method for parallel processing, should not put it in `FSUtils`, maybe we can add a similar tool method as `HoodieTimelineArchiver#deleteFilesParallelize` for hoodie instants, for example `HoodieTimelineArchiver#deleteInstantsParallelize`